src/proteus/dswx_hls.py

```diff
@@ -3437,47 +3670,61 @@ def generate_dswx_layers(input_list,
     # Note: The browse image will be always be saved as a separate png file;
     # it will not included in the combined `output_file`.
     if output_browse_image:
 
-        # If the `output_interpreted_band` was generated,
-        # convert that to the browse image
-        if output_interpreted_band:
-            src_geotiff_filename = output_interpreted_band
-        else:
-            # If the `output_interpreted_band` geotiff was not generated,
-            # we'll need to make it as a temporary file, and then convert
-            # that to the browse image.
-
-            # Create the source image as a geotiff
-            # Reason: gdal.Create() cannot currently create .png files, so we
-            # must start from a GeoTiff, etc.
-            # Source: https://gis.stackexchange.com/questions/132298/gdal-c-api-how-to-create-png-or-jpeg-from-scratch
-            src_geotiff_filename = os.path.join(scratch_dir,"dswx_tmp.tif")
-            save_dswx_product(masked_dswx_band,
-                            src_geotiff_filename,
-                            dswx_metadata_dict,
-                            geotransform,
-                            projection,
-                            scratch_dir=scratch_dir)
-
-            # add the temp file to the list to be removed at the end
-            temp_files_list += [src_geotiff_filename]
-
-        # Convert the geotiff to a PNG to create the browse image
-        geotiff2png(src_geotiff_filename=src_geotiff_filename,
+        # Build the browse image
+        # Create the source image as a geotiff
+        # Reason: gdal.Create() cannot currently create .png files, so we
+        # must start from a GeoTiff, etc.
+        # Source: https://gis.stackexchange.com/questions/132298/gdal-c-api-how-to-create-png-or-jpeg-from-scratch
+        wtr2_with_translucent_clouds = _compute_wtr2_with_translucent_clouds_array(
+                    unmasked_interpreted_water_layer=landcover_shadow_masked_dswx,  # WTR-2
+                    cloud_layer=cloud,
+                    flag_collapse_wtr_classes=FLAG_COLLAPSE_WTR_CLASSES,
+                    flag_no_snow_mask=flag_no_snow_mask_browse)
+
+        # Form color table
+        # Note: there is no need to account for `flag_no_snow_mask_browse` here.
+        # If `flag_no_snow_mask_browse` is False, then snow pixels would be set to
+        # `WTR_CLOUD_MASKED_SNOW` in _compute_wtr2_with_translucent_clouds_array(),
+        # and thus will appear as opaque snow.
+        # If `flag_no_snow_mask_browse` is True, then no pixels would be set to
+        # `WTR_CLOUD_MASKED_SNOW`, and thus no pixels would be colored as snow.
+        wtr2_with_translucent_clouds_ctable = _get_wtr2_translucent_clouds_ctable(
+            alpha=0.35,
+            flag_collapse_wtr_classes=FLAG_COLLAPSE_WTR_CLASSES)
+
+        # Save to intermediary .tif temp file to scratch directory
+        tmp_filename = "dswx_wtr2_trans_clouds_tmp.tif"
```

Comment:
Please, use `tempfile.NamedTemporaryFile()` for that.

Reply:
Will do.

Out of curiosity, what is the reasoning behind having some temp files use `tempfile.NamedTemporaryFile()`, but these others having hardcoded names?

```
    ../scratch_dir/copernicus_reprojected.tif
    ../scratch_dir/worldcover_reprojected_up_3.tif
```

Reply:
Ahh, that makes sense. `tempfile` it is for all. Thank you for updating those other filenames!